pkg/cmd/infra/deployer/deployer.go

```diff
@@ -189,17 +229,20 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 		// Scale the deployment down to zero.
 		retryWaitParams := kubectl.NewRetryParams(1*time.Second, 120*time.Second)
 		if err := d.scaler.Scale(candidate.Namespace, candidate.Name, uint(0), &kubectl.ScalePrecondition{Size: -1, ResourceVersion: ""}, retryWaitParams, retryWaitParams); err != nil {
-			glog.Errorf("Couldn't scale down prior deployment %s: %v", deployutil.LabelForDeployment(&candidate), err)
+			fmt.Fprintf(d.errOut, "error: Couldn't scale down prior deployment %s: %v\n", deployutil.LabelForDeployment(&candidate), err)
 		} else {
-			glog.Infof("Scaled down prior deployment %s", deployutil.LabelForDeployment(&candidate))
+			fmt.Fprintf(d.out, "--> Scaled older deployment %s down\n", candidate.Name)
 		}
 	}
 
+	if d.until == "start" {
+		return strategy.NewConditionReachedErr("Ready to start deployment")
```

Comment:
They're exceptions to the normal flow - it's handled at the top level.
Effectively we have defined a structured termination condition.

On Tue, May 10, 2016 at 11:55 AM, Michail Kargakis <notifications@github.com
> wrote:

> In pkg/cmd/infra/deployer/deployer.go
> <https://github.com/openshift/origin/pull/8787#discussion_r62698862>:
>
> >  		}
> >  	}
> >
> > +	if d.until == "start" {
> > +		return strategy.NewConditionReachedErr("Ready to start deployment")
>
> Umm, these conditions are not errors.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8787/files/a96fb629083f11e94f9496eb3fd5d5fa15ab9245#r62698862>
>